packages/app/contexts/TokenListContext.tsx

```diff
@@ -97,9 +122,49 @@ export const TokenListProvider = ({ children }: PropsWithChildren) => {
     setupTokenList();
   }, [address, setupTokenList]);
 
+  /**
+   * Once the tokenList is defined and we have an address
+   * connected, we read data from the blockchain using
+   * the Provider through the Multicall SC to batch fetch
+   * all the ERC-20 token balances for the connected wallet
+   * and the supported token list, then we desc. sort them
+   */
+  useEffect(() => {
+    if (address) {
+      provider
+        .call({
+          to: MULTICALL_ADDRESS,
+          data: multicallInterface.encodeFunctionData("aggregate", [callArray]),
+        })
```

Comment:
Maybe we should be using https://wagmi.sh/core/actions/multicall here

Reply:
This will probably let us free from `provider`

Reply:
Never used it, will try it out and see the results :D

Reply:
Made this change after updating `viem` and `wagmi` deps, thanks a lot for the off-line guidance messages :D